bandwidth-file-spec.txt

```diff
@@ -479,95 +621,260 @@
 
 2.4.2.1. Simple Bandwidth Scanner
 
-  sbws RelayLines may contain these keys:
+  sbws RelayLines contain these keys:
 
-    "node_id=" hexdigest SP
+    "node_id=" hexdigest
 
       As above.
 
-    "bw=" Bandwidth SP
+    "bw=" Bandwidth
 
       As above.
 
-    "nick=" nickname SP
+    "nick=" nickname
 
       [Exactly once.]
 
       The relay nickname.
 
-    "rtt=" Int SP
+      Torflow also has a "nick=" KeyValue.
 
-      [Exactly once.]
+    "rtt=" Int
+
+      [Zero or one time.]
 
       The Round Trip Time in milliseconds to obtain 1 byte of data.
 
-    "time=" DateTime SP
+      This KeyValue was added in version 1.1.0 of this specification.
+      It became optional in version 1.3.0 or 1.4.0 of this specification.
+
+    "time=" DateTime
 
       [Exactly once.]
 
       The date and time timestamp in ISO 8601 format and UTC time zone
       when the last bandwidth was obtained.
 
-    "success=" Int SP
+      This KeyValue was added in version 1.1.0 of this specification.
+      The Torflow equivalent is "measured_at=".
+
+    "success=" Int
 
       [Zero or one time.]
 
       The number of times that the bandwidth measurements for this relay were
       successful.
 
-    "error_circ=" Int SP
+      This KeyValue was added in version 1.1.0 of this specification.
+
+    "error_circ=" Int
 
       [Zero or one time.]
 
       The number of times that the bandwidth measurements for this relay
       failed because of circuit failures.
 
-    "error_stream=" Int SP
+      This KeyValue was added in version 1.1.0 of this specification.
+      The Torflow equivalent is "circ_fail=".
+
+    "error_stream=" Int
 
       [Zero or one time.]
 
       The number of times that the bandwidth measurements for this relay
       failed because of stream failures.
 
-    "error_misc=" Int SP
+      This KeyValue was added in version 1.1.0 of this specification.
+
+    "error_destination=" Int
```

Comment:
I thought we decided to don't have to include all the Lines KeyValues, but i'm probably wrong.

Reply:
We decided that we don't have to include all the Torflow KeyValues.

But if we want people to be able to use the sbws KeyValues, we need to tell them what they mean.
(Also, it makes the code easier to review and change.)